pkg/cmd/roachtest/test_runner.go

```diff
@@ -926,6 +933,30 @@ func (r *testRunner) runWorker(
 	}
 }
 
+// destroyCluster runs cluster destroy in a goroutine and adds 1 to the wait group.
+// if the cluster is nil, it just returns.
+// if the cluster is local, the cluster destroy is sequential.
+func (r *testRunner) destroyCluster(
+	ctx context.Context, clusterDestroyWg *sync.WaitGroup, c *clusterImpl, l *logger.Logger,
+) {
+	if c == nil {
+		// all good - nothing to destroy
+		return
+	}
+	if c.IsLocal() {
+		// local cluster - do not use goroutine as we generally recreate the cluster.
+		c.Destroy(context.Background(), closeLogger, l)
+		return
+	}
+	clusterDestroyWg.Add(1)
+	go func(ci *clusterImpl) {
+		defer clusterDestroyWg.Done()
+		// We use a context that can't be canceled for the Destroy().
+		ci.Destroy(context.Background(), closeLogger, l)
+		l.PrintfCtx(ctx, "cluster %s destroyed.", ci)
```

Comment:
Nit: this seems redundant. We already log completion in `cluster.go` (`doDestroy`).